salt/metalk8s/kubeadm/init/certs/ca.sls

```diff
@@ -0,0 +1,86 @@
+{%- from "metalk8s/map.jinja" import ca with context %}
+
+{%- set ca_server = salt['mine.get']('*', 'kubernetes_ca_server') %}
+
+{#- Check if we have no CA server or only current minion as CA #}
+{%- if not ca_server or ca_server.keys() == [grains['id']] %}
+
+include:
+  - .installed
+
+Create CA private key:
+  x509.private_key_managed:
+    - name: /etc/kubernetes/pki/ca.key
+    - bits: {{ ca.cert.private_key_size }}
+    - user: root
+    - group: root
+    - mode: 600
+    - makedirs: True
+    - dir_mode: 755
+    - require:
+      - pkg: Install m2crypto
+
+Generate CA certificate:
+  x509.certificate_managed:
+    - name: /etc/kubernetes/pki/ca.crt
+    - signing_private_key: /etc/kubernetes/pki/ca.key
+    - CN: kubernetes
+    - keyUsage: "critical digitalSignature, keyEncipherment, keyCertSign"
+    - basicConstraints: "critical CA:true"
+    - days_valid: {{ ca.cert.days_valid }}
+    - user: root
+    - group: root
+    - mode: 644
+    - makedirs: True
+    - dir_mode: 755
+    - require:
+      - x509: Create CA private key
+
+# TODO: Find a better way to advetise CA server
+Advertise CA in the mine:
+  module.wait:
+    - mine.send:
+      - func: 'kubernetes_ca_server'
+      - mine_function: test.ping
+    - watch:
+      - x509: Generate CA certificate
+
+Create CA salt signing_policies:
+  file.serialize:
+    - name: /etc/salt/minion.d/30-metalk8s_ca_signing_policies.conf
+    - user: root
+    - group: root
+    - mode: 644
+    - makedirs: True
+    - dir_mode: 755
+    - formatter: yaml
+    - dataset:
+        x509_signing_policies:
+          kube_apiserver_policy:
+            - minions: '*'
+            - signing_private_key: /etc/kubernetes/pki/ca.key
+            - signing_cert: /etc/kubernetes/pki/ca.crt
+            - keyUsage: "critical digitalSignature, keyEncipherment"
+            - extendedKeyUsage: "serverAuth"
+            - days_valid: {{ ca.signing_policy.days_valid }}
+
+Restart salt-minion:
```

Comment:
This kind of state should be something generic available upper in the formula (like metalk8s.salt.minion) and be included.

Especially with an ID like this, we risk collision and this is a "trick" (using bg and stuff) that can be missed by someone so I would rather a central place to control the restart of the minion and do a watch_in in this state

Reply:
Oh yes true I will create a ticket